uber/decorators.py

```diff
@@ -81,7 +81,17 @@ def _suffix_property_check(inst, name):
 def csrf_protected(func):
     @wraps(func)
     def protected(*args, csrf_token, **kwargs):
-        check_csrf(csrf_token)
+        try:
+            check_csrf(csrf_token)
+        except Exception as e:
+            s = str(e).lower()
+            if "csrf" in s:
+                message = "Missing CSRF token. Your session may have timed out. Please go back and try again."
+                log.error("CSRF Error: {}", str(e))
+                raise HTTPRedirect("../common/invalid?message={}", message)
```

Comment:
see below about only catching CSRFException here and removing the "if 'csrf' in s" statement here.